src/gc/gc.cpp

```diff
@@ -12392,6 +12392,11 @@ wait_full_gc_status gc_heap::full_gc_wait (GCEvent *event, int time_out_ms)
 
     if ((wait_result == WAIT_OBJECT_0) || (wait_result == WAIT_TIMEOUT))
     {
+#ifdef MULTIPLE_HEAPS
+        maxgen_percent = g_heaps[0]->fgn_maxgen_percent;
```

Comment:
> maxgen_percent = g_heaps[0]->fgn_maxgen_percent; [](start = 7, length = 49)

maxgen_percent was already read at the beginning of this method.

Reply:
oh I see your comment from above - is it because someone could register a new maxgen_percent?

Reply:
oh I see your comment from above - is it because the 1st time we read the value we observe its not set yet?

Reply:
The operation may be canceled while we're waiting, and `CancelFullGCNotification` works by setting `fgn_maxgen_percent`.

Reply:
ahh right; we still wanna return the right status so we need to re-read. makes sense.

Reply:
btw we could also write the code this way (like what the other places are doing) so you don't have to duplicate the code:
```
#ifdef MULTIPLE_HEAPS
    gc_heap* hp = gc_heap::g_heaps[0];
#else
    gc_heap* hp = pGenGCHeap;
#endif //MULTIPLE_HEAPS

if (hp->fgn_maxgen_percent == 0)
```